doc/guides/dts/index.rst

```diff
@@ -1,177 +1,249 @@
+
+Devicetree source code files are written in the DTS format, which is defined in
+the `Devicetree specification`_. They usually have ``.dts`` or ``.dtsi`` (for
+Devicetree Source Include) extensions. Zephyr's build system looks for a file
+named :file:`BOARD.dts` in the board definition directory; this file contains
+the board's base devicetree. See :ref:`dt_k6x_example` for real-world examples.
+
+The build system combines the board's DTS with additional input files called
+*overlays* to produce a final devicetree source file. Overlays are also written
+in the DTS format, but have a :file:`.overlay` extension to make it clear that
+they're overlays. You can specify the overlay files to use at build time using
+the :makevar:`DTC_OVERLAY_FILE` CMake variable described in
+:ref:`important-build-vars`. The build system also looks for device tree
+overlays in several locations by default; see :ref:`application_dt` for the
+list.
+
+Overlays can be used to add or delete nodes from the tree, or to modify node
+properties and their values. Along with Kconfig, devicetree overlays let you
+reconfigure the kernel and device drivers without modifying their source code.
+
+Before they are combined, the C preprocessor is run on :file:`BOARD.dts` and any
+overlays. This allows these files to use C macros and include directives.
+
+The *devicetree compiler* tool, ``dtc``, is used to combine :file:`BOARD.dts`
+with any overlays into a final :file:`BOARD.dts_compiled` file in the
+application build directory.  :file:`BOARD.dts_compiled` is the final
+devicetree source. It is also in DTS format. More information on ``dtc`` can be
+found in the `dtc source code repository`_.
+
+.. _dtc source code repository: https://git.kernel.org/pub/scm/utils/dtc/dtc.git
+
+Next, :file:`BOARD.dts_compiled` and the set of :ref:`bindings` are passed
+to Python scripts in :zephyr_file:`scripts/dts/` to generate C
+definitions. These can be included via the ``generated_dts_board.h`` header
+file, which the build system places on the C preprocessor include path. Do not
+include the generated C headers in the build directory directly -- use
+``generated_dts_board.h`` instead. A file named
+:file:`generated_dts_board.conf` is also placed in the build directory; it is
+mostly useful for inspecting the results of Zephyr's DTS scripts.
+
+Zephyr device drivers typically use information from ``generated_dts_board.h``
+to statically allocate and initialize :ref:`struct device <device_struct>`
+instances. Property values from ``generated_dts_board.h`` are usually stored in
+ROM in the value pointed to by a ``device->config->config_info`` field. For
+example, a ``struct device`` corresponding to an I2C peripheral would store the
+peripheral address in its ``reg`` property there.
+
+Application source code with a pointer to the ``struct device`` can then pass
+it to driver APIs in :zephyr_file:`zephyr/include/drivers`. These API functions
```

Comment:
Nit: `zephyr/include/drivers/`, for consistency.

Reply:
>  Nit: zephyr/include/drivers/, for consistency.

Whoops, that's a bug. It should be include/drivers, not zephyr/include/drivers. I'll add the slash.